### PR TITLE
Implementation of Gregicality Multiblocks

### LINKED ISSUE #311: Deepslate ores unable to be processed in ATM9
### GregTech CEu Version

1.20.1 - 1.0.11

### Recipe Viewer Installed

JEI

### Environment

Dedicated Server

### Cross-Mod Interaction

Yes

### Expected Behavior

recipes available to process deepslate ores

### Actual Behavior

no recipes to process deepslate ores available

### Steps to Reproduce

1) get deepslate ores from occultism dimensional mineshafts
2) check recipes for using the ores in JEI
3) no results

### Additional Information

im only able to get these ores from the occultism dimensional mineshaft, but figured id report it anyway

### LINKED ISSUE #330: KJS Multiblocks cannot overclock
### GregTech CEu Version

1.0.11.a

### Recipe Viewer Installed

EMI

### Environment

Singleplayer

### Cross-Mod Interaction

Yes

### Expected Behavior

A custom recipe set to consume 16EU/t should overclock in a custom multiblock using a HV energy hatch to 256EU/t

### Actual Behavior

The multiblock only consumes 16EU/t despite the overclock being possible

### Steps to Reproduce

Define a custom multiblock using KubeJS with a custom recipe type, add a recipe to the multiblock with an EU/t many tiers below the maximum capacity of the machine, then supply items and power

### Additional Information

I have set up parallel recipe support by loading GTRecipeModifiers, but multiblocks still do not overclock even without parallel logic. I've also tried using the overclock logic of the EBF, Multi Smelter etc, but the multiblock does not start the recipe if those classes are loaded

### LINKED ISSUE #367: Creative mode crash when trying to access GTCE Tabs
### GregTech CEu Version

1.19.2-1.0.11.b-build_173.jar 

### Recipe Viewer Installed

None

### Environment

Single Player

### Cross-Mod Interaction

Unsure

### Expected Behavior

Load into world and get into creative mode to spawn Gregtech stuff (or anything else that may be on the same page as GT's creative tabs)

### Actual Behavior

When in creative inventory, then trying to access a "Page" with GT Creative tabs, the game crashes.

### Steps to Reproduce

Start game
Load world
Go into creative mode
Open creative inventory and attempt to move to the next page of tabs
Game crashes.

### Additional Information

I had to have notenoughcrashes-4.2.1+1.19.2-forge.jar enabled to get a crash report as the log file and debug files don't say much about the crash. 

Without notenoughcrashes-4.2.1+1.19.2-forge.jar the game just hangs up when trying to access a page of GT tabs, and never closes, crashes, responds, etc.

Here are the files anyways, so that perhaps something might help.
[latest.log](https://github.com/GregTechCEu/GregTech-Modern/files/12614805/latest.log)

[debug.log](https://github.com/GregTechCEu/GregTech-Modern/files/12614807/debug.log)

[crash-2023-09-14_21.42.37-client.txt](https://github.com/GregTechCEu/GregTech-Modern/files/12614808/crash-2023-09-14_21.42.37-client.txt)

### PR DESCRIPTION
**Pull request by Rundas01**
shoutouts to Screret for the Parallel Hatches

### CODE REVIEW

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+                        case 5 -> "Elite";
+                        case 6 -> "Master";
+                        case 7 -> "Ultimate";
+                        case 8 -> "Super";
+                        default -> "Simple"; // Should never be hit.
+                    } + " Parallel Control Hatch")
+                    .rotationState(RotationState.ALL)
+                    .abilities(PartAbility.PARALLEL_HATCH)
+                    .workableTieredHullRenderer(GTCEu.id("block/machines/parallel_hatch_mk" + (tier - 4)))
+                    .tooltips(Component.translatable("gtceu.parallelhatch_mk"+tier))
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]: this is a typo, no?

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java [hunk 2]
```diff
@@ -0,0 +1,732 @@
...
+            .workableCasingRenderer(GTCEu.id("block/casings/gcym/secure_maceration_casing"),
+                    GTCEu.id("block/multiblock/gcym/large_maceration_tower"), false)
+            .compassSections(GTCompassSections.TIER[IV])
+            .compassNodeSelf()
+            .register();
+
+    public final static MultiblockMachineDefinition LARGE_CHEMICAL_BATH = REGISTRATE.multiblock("large_chemical_bath", WorkableElectricMultiblockMachine::new)
+            .langValue("Large Chemical Bath")
+            .tooltips(Component.translatable("gtceu.multiblock.parallelizable.tooltip"))
+            .tooltips(Component.translatable("gtceu.machine.available_recipe_map_2.tooltip", Component.translatable("gtceu.recipemap.ore_washer"), Component.translatable("gtceu.recipemap.chemical_bath")))
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]: these ain't the right lang keys, use the preexisting ones like `gtceu.ore_washer`, that you can get from `CHEMICAL_BATH_RECIPES.registryName.toLanguageKey()`

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java [hunk 3]
```diff
@@ -536,6 +538,32 @@ public class GTRecipeTypes {
...
+    public final static GTRecipeType BLAST_ALLOY_RECIPES = register("blast_alloy_smelter", ELECTRIC).setMaxIOSize(9, 0, 3, 1).setEUIO(IO.IN)
+            .setProgressBar(GuiTextures.PROGRESS_BAR_ARROW, LEFT_TO_RIGHT)
+            .setSlotOverlay(false, false, false, GuiTextures.FURNACE_OVERLAY_1)
+            .setSlotOverlay(false, false, true, GuiTextures.FURNACE_OVERLAY_1)
+            .setSlotOverlay(false, true, false, GuiTextures.FURNACE_OVERLAY_2)
+            .setSlotOverlay(false, true, true, GuiTextures.FURNACE_OVERLAY_2)
+            .setSlotOverlay(true, true, false, GuiTextures.FURNACE_OVERLAY_2)
+            .setSlotOverlay(true, true, true, GuiTextures.FURNACE_OVERLAY_2)
+            .setSound(GTSoundEntries.ARC);
+
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GTRecipeTypes.java]: these shouldn't be here i think

File: common/src/main/java/com/gregtechceu/gtceu/data/lang/LangHandler.java [hunk 4]
```diff
@@ -105,6 +105,47 @@ public static void init(RegistrateLangProvider provider) {
...
+        provider.add("gtceu.recipemap.autoclave", "Autoclave");
+        provider.add("gtceu.recipemap.bender", "Bender");
+        provider.add("gtceu.recipemap.hammer", "Forge Hammer");
+        provider.add("gtceu.recipemap.press", "Forming Press");
+        provider.add("gtceu.recipemap.heater", "Fluid Heater");
+        provider.add("gtceu.recipemap.cutter", "Cutter");
+        provider.add("gtceu.recipemap.lathe", "Lathe");
+        provider.add("gtceu.recipemap.canner", "Canner");
+        provider.add("gtceu.recipemap.solidifier", "Fluid Solidifier");
+
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/data/lang/LangHandler.java]: use the autogenerated language keys. like `gtceu.solidifier`

**Review by screret — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java (hunk 1 above)
Comment (screret) [reply common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]: also for the lang key you could do `"gtceu.tooltip.parallel_hatch_mk" + (tier - 4)`

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java [hunk 5]
```diff
@@ -383,7 +383,7 @@ public static void init() {}
     public final static MultiblockMachineDefinition BLAST_ALLOY_SMELTER = REGISTRATE.multiblock("alloy_blast_smelter", CoilWorkableElectricMultiblockMachine::new)
             .langValue("Alloy Blast Smelter")
             .tooltips(Component.translatable("gtceu.multiblock.parallelizable.tooltip"))
-            .tooltips(Component.translatable("gtceu.machine.available_recipe_map_1.tooltip", Component.translatable("gtceu.recipemap.blast_alloy")))
+            .tooltips(Component.translatable("gtceu.machine.available_recipe_map_1.tooltip", Component.translatable("gtceu.blast_alloy")))
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]: i think this lang key should be `gtceu.alloy_blast_smelter`?

**Review by screret — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java [hunk 6]
```diff
@@ -0,0 +1,732 @@
...
+            .workableCasingRenderer(GTCEu.id("block/casings/gcym/vibration_safe_casing"),
+                    GTCEu.id("block/multiblock/gcym/large_sifting_funnel"), false)
+            .compassSections(GTCompassSections.TIER[IV])
+            .compassNodeSelf()
+            .register();
+
+    public final static MultiblockMachineDefinition BLAST_ALLOY_SMELTER = REGISTRATE.multiblock("alloy_blast_smelter", CoilWorkableElectricMultiblockMachine::new)
+            .langValue("Alloy Blast Smelter")
+            .tooltips(Component.translatable("gtceu.multiblock.parallelizable.tooltip"))
+            .tooltips(Component.translatable("gtceu.machine.available_recipe_map_1.tooltip", Component.translatable("gtceu.blast_alloy_smelter")))
```
Comment (screret) [inline common/src/main/java/com/gregtechceu/gtceu/common/data/GCyMMachines.java]:
```suggestion
            .tooltips(Component.translatable("gtceu.machine.available_recipe_map_1.tooltip", Component.translatable("gtceu.alloy_blast_smelter")))
```

**Review by screret — APPROVED** [review]

**Review by screret — DISMISSED** [review]

**Review by whatthedrunk — APPROVED** [review]

**Review by TheAlaskanPhysicist — APPROVED** [review]
IT WORKED

**Pull request merged**